docs/source/notebooks/tpt.ipynb

```diff
@@ -0,0 +1,779 @@
+    "sim.add_barrier((8, 7), (9, 7), weight=5.)\n",
+    "\n",
+    "sim.add_barrier((0, 2), (2, 2), weight=5.)\n",
+    "sim.add_barrier((2, 0), (2, 1), weight=5.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can simulate a trajectory on this grid by specifying a starting point and a number of simulation steps. The effective length of the trajectory might be lower than the number of simulation steps as the simulation stops if the state is `home` or `bar`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start = (7, 2)\n",
+    "walk = sim.walk(start=start, n_steps=250, seed=40)\n",
+    "print(\"Number of steps in the walk:\", len(walk))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The trajectory can be visualized with a few helper functions attached to the simulator:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib as mpl\n",
+    "import matplotlib.pyplot as plt\n",
+    "from mpl_toolkits.axes_grid1 import make_axes_locatable\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(10, 10))\n",
+    "\n",
+    "ax.scatter(*start, marker='*', label='Start', c='cyan', s=150, zorder=5)\n",
+    "sim.plot_path(ax, walk)\n",
+    "handles, labels = sim.plot_2d_map(ax)\n",
+    "ax.legend(handles=handles, labels=labels);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, the darker red squares denote hard barriers, the lighter red squares denote barriers which can be crossed. So one could imagine, that both home and bar are on a hill.\n",
+    "\n",
+    "The simulator internally holds a [MarkovStateModel](../api/generated/sktime.markov.msm.MarkovStateModel.rst#sktime.markov.msm.MarkovStateModel) on which one can call [reactive_flux(A, B)](../api/generated/sktime.markov.msm.MarkovStateModel.rst#sktime.markov.msm.MarkovStateModel.reactive_flux). This computes some quantities related to the ensemble of reactive trajectories between sets of states $A$ and $B$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Internal Markov state model with {sim.msm.n_states} states to cover the {sim.grid_size} grid.\")\n",
+    "print(f\"Compute reactive flux from A={sim.home_state} to B={sim.bar_state}\")\n",
+    "flux = sim.msm.reactive_flux(sim.home_state, sim.bar_state)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The committor\n",
+    "\n",
+    "One of the questions that can be answered with TPT is: If the man is at some state $(i,j)$, what is the probability that he reaches the bar before home? If he is already already at the bar, the probability is $1$, i.e., \n",
```

Comment:
the state gets 2 dimensions because i its its index in x and j is its index in y? i feel like i often see notation i --> j so had to read twice

Reply:
Yeah internally it is treated as a linear sequence of states of course, but each state maps to a 2-dimensional coordinate on the grid. That is why i thought it might be easier to work with grid coordinates directly as they can be related to the plots.